### PR TITLE
Update publishing of status messages over MQTT

### PR DESCRIPTION
**Pull request by vetlek**

### CODE REVIEW

**Comment by vetlek** [comment]
Closes #266

File: src/isar/config/settings.py [hunk 1]
```diff
@@ -192,18 +192,21 @@ class Settings(BaseSettings):
...
 
-    @validator("ISAR_STATE", "ISAR_MISSION", "ISAR_STEP_STATUS", pre=True, always=True)
+    ISAR_STEP: str = Field(default="step")
+
+    @validator(
+        "ISAR_STATE", "ISAR_MISSION", "ISAR_TASK", "ISAR_STEP", pre=True, always=True
+    )
     def prefix_isar_topics(cls, v, values):
-        return f"{values['ROBOT_ID']}/{v}"
+        return f"isar/{values['ROBOT_ID']}/{v}"
```
Comment (GodVenn) [inline src/isar/config/settings.py:209]: I really like the implementation of the prefixed isar to the topics!
Maybe we should change the naming of these config variables to make it clear they are MQTT topics?
Suggestion:
`ISAR_TASK` => `TOPIC_ISAR_TASK`

File: src/isar/state_machine/states/finalize.py [hunk 2]
```diff
@@ -36,3 +39,11 @@ def _log_state_transitions(self):
...
 
         self.logger.info(f"State transitions:\n  {state_transitions}")
+
+    def _update_tasks(self):
+        for task in self.state_machine.current_mission.tasks:
+            for step in task.steps:
+                if step.status == StepStatus.NotStarted:
+                    step.status = StepStatus.Cancelled
+            if task.status == StepStatus.NotStarted:
+                task.status = StepStatus.Cancelled
```
Comment (GodVenn) [inline src/isar/state_machine/states/finalize.py:49]: Should we already now introduce seperate enums for `StepStatus` and `TaskStatus`?

File: src/isar/state_machine/states/monitor.py [hunk 3]
```diff
@@ -39,8 +39,7 @@ def start(self):
 
     def stop(self):
         if self.state_machine.mqtt_client:
-            self.state_machine.publish_step_status()
-            self.state_machine.publish_mission()
+            self.state_machine.publish_task_status()
```
Comment (GodVenn) [inline src/isar/state_machine/states/monitor.py:42]: Should we also publish the initial mission status here?

**Review by GodVenn — COMMENTED** [review]
Nice!
Please see the comments for discussion.

File: src/isar/state_machine/state_machine.py [hunk 4]
```diff
@@ -266,31 +273,67 @@ def stop_mission(self):
...
         self.logger.info(message)
         if not failure:
+            self.current_mission.status = MissionStatus.Cancelled
             self.mission_in_progress = False
+            for task in self.current_mission.tasks:
+                for step in task.steps:
+                    if step.status == StepStatus.NotStarted:
+                        step.status = StepStatus.Cancelled
+                if task.status == TaskStatus.NotStarted:
+                    task.status = TaskStatus.Cancelled
```
Comment (GodVenn) [inline src/isar/state_machine/state_machine.py:283]: Is the status changed to cancelled only if it's not started?
If a task is in progress when the mission is cancelled, should it not also be marked as cancelled?

**Review by GodVenn — COMMENTED** [review]
Please see the comments

File: src/isar/models/mission/status.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    Completed: str = "completed"
+
+
+class TaskStatus(str, Enum):
+    NotStarted: str = "not_started"
+    InProgress: str = "in_progress"
+    PartiallySuccessful: str = "partially_successful"
+    Failed: str = "failed"
+    Cancelled: str = "cancelled"
+    Completed: str = "completed"
```
[the remark this answers is not in the document]
Comment (vetlek) [reply src/isar/models/mission/status.py:19]: Agree

File: src/isar/state_machine/state_machine.py (hunk 4 above)
Comment (vetlek) [reply src/isar/state_machine/state_machine.py:283]: Good catch

File: src/isar/state_machine/state_machine.py [hunk 6]
```diff
@@ -272,13 +273,14 @@ def stop_mission(self):
         self.queues.stop_mission.output.put(deepcopy(message))
         self.logger.info(message)
         if not failure:
+            self.current_mission.status = MissionStatus.Cancelled
             self.mission_in_progress = False
             for task in self.current_mission.tasks:
                 for step in task.steps:
-                    if step.status == StepStatus.NotStarted:
+                    if step.status == [StepStatus.NotStarted, StepStatus.InProgress]:
```
Comment (GodVenn) [inline src/isar/state_machine/state_machine.py:280]:
``` suggestion
if step.status in [StepStatus.NotStarted, StepStatus.InProgress]:
```

**Review by GodVenn — CHANGES_REQUESTED** [review]

**Review by GodVenn — APPROVED** [review]
LGTM

**Pull request merged**